src/averageAges.js

```diff
@@ -55,10 +80,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothersWithChildren = people.reduce((arr, person) => {
+    const childrenBorn = people.filter((child) => {
+      const hasMother = person.name === child.mother;
+
+      return onlyWithSon ? child.sex === 'm' && hasMother : hasMother;
+    }).map((child) => child.born);
+
+    if (!childrenBorn.length) {
+      return arr;
+    }
+
+    return [...arr, {
+      motherBorn: person.born,
+      childrenBorn,
+    }];
+  }, []);
+
+  const ages = mothersWithChildren.reduce((arrAges, mother) => {
+    const agesDiff = mother.childrenBorn.map(
+      (born) => born - mother.motherBorn);
+
+    return [...arrAges, ...agesDiff];
+  }, []);
+
+  const averageAge = ages.reduce((sum, age) => sum + age, 0) / ages.length;
```

Comment:
Instead of manually rounding the average age to two decimal places, consider using the toFixed method. E.g., averageAge.toFixed(2).